sdetools/sdelib/ldap_module.py

```
@@ -0,0 +1,269 @@
+import ldap
+import ldap.modlist as modlist
+import re
+from sdetools.sdelib import log_mgr
+from sdetools.sdelib.commons import Error
+from sdetools.sdelib.interactive_plugin import PlugInExperience
+from sdetools.sdelib.restclient import APIError
+import sdeapi
+
+logger = log_mgr.mods.add_mod(__name__)
+
+
+class LDAPConnectorError(Error):
+    pass
+
+
+class LDAPConnector(object):
+
+	def __init__(self, config):
+		self.config = config
+		self._add_config_options()
+		self.sde_plugin = sdeapi.APIBase(self.config)
+
+	def _add_config_options(self):
+		""" Adds LDAP specific config options to the config file"""
+		self.config.opts.add("ldap_endpoint_url", "endpoint of LDAP server")
+		self.config.opts.add("remote_method", "protocol of LDAP connection")
+		self.config.opts.add("bind_dn", "DN of LDAP user")
+		self.config.opts.add("bind_password", "password of LDAP user")
+		self.config.opts.add("group_mapping", "mapping of AD/LDAP Groups to SD Elements")
+		self.config.opts.add("filter", "filter for users and groups during synchronization")
+		self.config.opts.add("deactivation", "toggle to automatically deactivate groupless users in SDE")
+
+	def _compute_base_dn(self):
+		DNsplit = re.split(",", self.config["bind_dn"])
+		basedn = ""
+		for item in DNsplit:
+			if re.match("dc=", item):
+				basedn += "," + item
+		return basedn[1:]
+
+	def initialize(self):
+		"""Sanity Check Mappings and post init processing"""
+		self.config.process_json_dict("filter")
+		self.config.process_json_dict("group_mapping")
+		self.group_translate(self.config["filter"]["groups"])
```

Correct. It's for validation there because the thought process is that if it fails, it fails early rather than mid sync, leaving a partial sync.